src/test/java/org/mule/extension/socket/CustomProtocolRefTestCase.java

```diff
@@ -10,23 +10,34 @@
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.not;
 import static org.hamcrest.Matchers.nullValue;
+
 import org.mule.extension.socket.api.socket.tcp.TcpProtocol;
 
 import org.junit.Test;
 
+import javax.inject.Inject;
+import javax.inject.Named;
+
 public class CustomProtocolRefTestCase extends SocketExtensionTestCase {
 
+  @Inject
+  @Named("myProtocolRef")
+  private TcpProtocol protocol;
+
+  @Override
+  protected boolean doTestClassInjection() {
```

Comment:
I wonder why the default is not true... is there any case when we don't want some instance to be injected?